vcs/gitcmd/repo.go

```diff
@@ -962,6 +962,31 @@ func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, er
 	return committers, nil
 }
 
+func (r *Repository) ListFiles(at vcs.CommitID) ([]string, error) {
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if at == "" {
+		at = "HEAD"
+	}
+
+	cmd := exec.Command("git", "ls-files", "--with-tree", string(at))
```

Comment:
Please explain why is `git ls-tree` more direct than `git ls-files`, since the former command outputs a lot more information which will unnecessarily need to be parsed and discarded before returning the file names (which is what the latter command outputs).